### DISCUSSION TITLE
Icon command

### DISCUSSION THREAD

**Discussion by flatcap** [neomutt/neomutt#3253] [Development]
This discussion is born of @dcpurton's idea for an `icon` command in #3244.

### Status Characters

NeoMutt has 5 config variables that allow the user to customise the Index/Status bar.

- [`$crypt_chars`](https://neomutt.org/guide/reference#crypt-chars) - User-configurable crypto flags: signed, encrypted etc.
- [`$flag_chars`](https://neomutt.org/guide/reference#flag-chars) - User-configurable index flags: tagged, new, etc
- [`$from_chars`](https://neomutt.org/guide/reference#from-chars) - User-configurable index flags: to address, cc address, etc
- [`$status_chars`](https://neomutt.org/guide/reference#status-chars) - Indicator characters for the status bar
- [`$to_chars`](https://neomutt.org/guide/reference#to-chars) - Indicator characters for the 'To' field in the index

They can include Emojis or [Nerd Font](https://www.nerdfonts.com/) characters.

Currently, the Compose Dialog is getting a make-over.
Adding a new config variable `$attach_chars` would be useful.

However we need something more flexible, something that can handle arbitrary attachment types.

### Icon Command

Proposed format:

```sh
# Define an icon
icon type sub-type char [fallback-char]
```

These examples use indentation to avoid very repetitive lines.

**Config**: Attachment status indicators

```
icon attachment
        inline          📄  I 
        attachment      📎  A 
        countable       ⑫  Q 
        tagged          ✅  * 
        convert         🔄  c 
        deleted         ❌  d 
        unlink          ⏳  - 
```

**Config**: Attachment mime types

```
icon mime-types
        multipart/alternative   ↔️ A
        multipart/related       👪 R
        multipart/mixed         🔀 M
        multipart/multilingual  🌍 L
        text/.*                 📄
        image/.*                🖼️
        message/rfc822          ✉️
```

**Display**: Compose

```
⏳ 📄  1  ↔️ <no description>                            [multipart/alternative]
   📎  2  ├─>📄 <no description>                                    [text/plain]
⏳ 📎  3  └─>🌍 <no description>                                     [text/html]
   📎  4  🖼️ cat.gif                                                 [image/gif]
   📎  5  📹 dog.avi                                               [image/video]
```

**Comment by gahr** [comment]
When would `fallback-char` be used?

**Reply by flatcap** [reply]
There's a config var: `$ascii_chars` which "gracefully" uses ASCII chars instead of Unicode line-drawing chars.
I can't remember whether ncurses is smart enough to know whether it can draw lines.

If we had a fallback char, then one "theme" config file could define everything.
